feat: improve dockerfile verify subcommand

BobyMCbobs opened the pull request:
support images resolved from ENV, ARG and COPY --from

resolves: #3260

<!--
Thanks for opening a pull request! Please do not just delete this text.  The three fields below are mandatory.

Please remember to:
- This PR requires an issue. If it is a new feature, the issue should proceed the PR and will have allowed sufficent time for discussions to take place. Please use
issue tags such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ".
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!

Thank you :)
-->

#### Summary

`cosign dockerfile verify` does not interpolate variables in `FROM` statements, instead just panics.
Adds the ability to load key-value pairs from `ENV` and `ARG` to interpolate in `FROM` and `COPY --from` statements.

#### Release Note

Improved discoverability for images in Dockerfiles

cpanato commented on cmd/cosign/cli/dockerfile/verify_test.go:
```diff
@@ -71,21 +71,68 @@ func TestGetImagesFromDockerfile(t *testing.T) {
 			FROM gcr.io/$TEST_IMAGE_REPO_PATH/${TEST_SUBREPO}:latest
 			RUN script2
 			FROM --platform=linux/amd64 gcr.io/${TEST_IMAGE_REPO_PATH}/$TEST_RUNTIME_SUBREPO
+            COPY --from=gcr.io/someorg/someimage /etc/config /app/etc/config
```
This looks a bit off, can you check the indentation?

BobyMCbobs replied:
fixed in 76858d7

cpanato reviewed (changes requested):
lgtm
just a small nit

thanks for adding tests as well

BobyMCbobs commented:
> lgtm just a small nit

@cpanato , updated the indent and fixed the tests in https://github.com/sigstore/cosign/commit/76858d74da011ad2792ff1133934a490fd8825d6
 
> thanks for adding tests as well

Absolutely!

cpanato reviewed (approved):
lgtm
thank you

hectorj2f reviewed (approved):
lgtm

haydentherapper reviewed (approved):
Very cool! Thanks for the tests too!

BobyMCbobs commented:
Wooohoooo! Thank you for the merge @haydentherapper @hectorj2f @cpanato!
